### ISSUE TITLE
Flow generic arrow function with multi line parameters and implied return indentation detection inconsistency

### ISSUE THREAD

**Issue by ahumeau** [eslint/eslint#10941]
**Tell us about your environment**

* **ESLint Version:** `5.6.1`
* **Node Version:** `8.12.0`
* **npm Version:** `6.4.1`

**What parser (default, Babel-ESLint, etc.) are you using?**
`Babel-ESLint 10.0.1`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// .eslintrc.js
module.exports = {
  parser: 'babel-eslint',
  rules: {
    indent: ['error', 2],
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// index.js


const VALID_genericSingleLineParams = <T>(bar: T[]) => (
  bar
)


const VALID_notGenericMultiLineParams = (
  bar
) => (
  bar
)


const INVALID_genericMultiLineParams = <T>(
  bar: T[]
) => (
  bar
)
```

```bash
./node_modules/.bin/eslint index.js
```

**What did you expect to happen?**

I expected ESLint not to detect any error, i.e that the expected indentation for each function declaration's return value to be the same.

**What actually happened? Please include the actual, raw output from ESLint.**
```
index.js
  16:1  error  Expected indentation of 4 spaces but found 2  indent
  17:1  error  Expected indentation of 2 spaces but found 0  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

**N.B.** I first opened an issue with eslint-plugin-flowtype [here](https://github.com/gajus/eslint-plugin-flowtype/issues/360) where I was told the issue comes from ESLint

**Comment by kaicataldo** [comment]
Since ESLint is aware of type annotations (as defined by [Estree](https://github.com/estree/estree/blob/master/extensions/type-annotations.md)), it does sound like this is something that should be fixed in the `indent` rule itself.
